File: eCinema/eCinema.Core/Entities/Movie.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace eCinema.Core.Entities
{
    public class Movie
    {
        public int ID { get; set; }
        public string Title { get; set; } = null!;
        public string Description { get; set; } = null!;
        public string Author { get; set; } = null!;
        public int ReleaseYear { get; set; }
        public int Duration { get; set; }
        public byte[]? Photo { get; set; } = null!;

        [ForeignKey(nameof(Language))]
        public int? LanguageId { get; set; }
        public Language? Language { get; set; } 

        [ForeignKey(nameof(Production))]
        public int? ProductionId { get; set; }
        public Production? Production { get; set; } 

        public ICollection<MovieGenre> Genres { get; set; } = new List<MovieGenre>();
        public ICollection<MovieActor> Actors { get; set; } = new List<MovieActor>();
        public ICollection<Show> Shows { get; set; } = null!;
    }
}

File: eCinema/UI/ecinema_mobile/lib/models/movie.dart
import 'package:ecinema_mobile/models/language.dart';
import 'package:ecinema_mobile/models/movie_actor.dart';
import 'package:ecinema_mobile/models/movie_genre.dart';
import 'package:ecinema_mobile/models/production.dart';
import 'package:json_annotation/json_annotation.dart';

part 'movie.g.dart';

@JsonSerializable()
class Movie {
  int? id;
  String? title;
  String? description;
  String? author;
  int? duration;
  int? releaseYear;
  int? productionId;
  Production? production;
  int? languageId;
  Language? language;
  List<MovieGenre>? genres;
  List<MovieActor>? actors;
  String? photo;

  Movie(this.id, this.title, this.description, this.releaseYear, this.productionId, this.languageId, this.photo,
      this.genres, this.actors);

  factory Movie.fromJson(Map<String, dynamic> json) => _$MovieFromJson(json);

  Map<String, dynamic> toJson() => _$MovieToJson(this);
}
